src/client/datascience/notebook/vscodeNotebookController.ts

```diff
@@ -129,11 +133,12 @@ export class VSCodeNotebookController implements Disposable {
     public async updateNotebookAffinity(notebook: NotebookDocument, affinity: NotebookControllerAffinity) {
         this.controller.updateNotebookAffinity(notebook, affinity);
         // Only on CI Server.
-        if (IS_CI_SERVER) {
+        if (this.context.extensionMode === ExtensionMode.Test) {
             await this.commandManager.executeCommand('notebook.selectKernel', {
                 id: this.id,
                 extension: JVSC_EXTENSION_ID
             });
+            VSCodeNotebookController.kernelAssociatedWithDocument = true;
```

Comment:
I'm surprised our tests pass on CI, this is required since kernel push.
At least locally we end up with a prompt without this (I think its a timing issue, we have a 500ms delay in tests, probably that delay works on CI)